formstack-consents/src/main/scala/com/gu/identity/formstackconsents/LambdaService.scala

```diff
@@ -0,0 +1,33 @@
+package com.gu.identity.formstackconsents
+
+import com.gu.identity.formstackconsents.FormstackClient.FormstackConsent
+import com.gu.identity.globalConfig.DevConfig
+import com.typesafe.scalalogging.StrictLogging
+
+class LambdaService(config: DevConfig, formstackClient: FormstackClient, identityClient: IdentityClient) extends StrictLogging {
+
+  def getEmailConsentFromSubmission(submission: FormstackClient.FormstackSubmission): Either[Throwable, FormstackConsent] = {
+    val emailField = submission.data.find(field => field._2.label.toLowerCase().contains("email address") && field._2.consentType == "email")
+    emailField match {
+      case Some(consent) => Right(consent._2)
+      case None =>
+        val errorMessage = "Unable to find email address field in Formstack submission"
+        logger.error(errorMessage)
+        Left(new Throwable(errorMessage))
+    }
+  }
+
+  def sendConsentToIdentity(consent: FormstackConsent, newsletter: Newsletter): Either[Throwable, Unit] = {
+    identityClient.sendConsentToIdentity(consent, newsletter)
+  }
+
+  def getConsentsAndSendToIdentity(newsletter: Newsletter): Either[Throwable, Unit] = {
```

Comment:
Open to comments on a better way to write this method

Reply:
After chatting we will try something like 
```
def processPageResponses(pageResponses: List[FormstackClient.FormstackResponse], newsletter: Newsletter): List[Either[Throwable,HttpResponse[String]]] = {
    val submissions: List[FormstackClient.FormstackSubmission] = pageResponses.flatMap(_.submissions)
    val emails: List[Either[Throwable, FormstackConsent]] = submissions.map(getEmailConsentFromSubmission)
    emails.map{ consentsOrErrors =>
      consentsOrErrors.flatMap(consent => identityClient.sendConsentToIdentity(consent, newsletter))
    }
  }
```